core/src/test/java/org/apache/iceberg/TestOverwrite.java

```diff
@@ -135,6 +135,49 @@ public void createTestTable() throws IOException {
     commit(table, table.newAppend().appendFile(FILE_0_TO_4).appendFile(FILE_5_TO_9), branch);
   }
 
+  @TestTemplate
+  public void deleteDataFilesProducesDeleteOperation() {
+    OverwriteFiles overwriteFiles = table.newOverwrite().deleteFile(FILE_A).deleteFile(FILE_B);
+
+    commit(table, overwriteFiles, branch);
+    Snapshot snap = latestSnapshot(table, branch);
```

Comment:
Nit: I guess you could just inline snap on 144

Reply:
done